### ISSUE TITLE
0.0.8 version brokes additional_cost template

### ISSUE THREAD

**Issue by mnuxx** [custom-components/nordpool#217]
2022-11-10 23:21:22.257 ERROR (MainThread) [homeassistant.helpers.template] Template variable error: 'None' has no attribute 'weekday' when rendering '{% set s = { "day": 37300, "night": 21400 } %} {% if now().weekday() in (5,6) %} {{s.night|float}} {% else %} {% if now().hour >= 7 and now().hour < 22%} {{s.day|float}} {% else %} {{s.night|float}} {% endif %} {% endif %}'
2022-11-10 23:21:22.267 ERROR (MainThread) [homeassistant.util.logging] Exception in check_stuff when dispatching 'nordpool_update': ()
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/usr/local/lib/python3.10/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<template>", line 1, in top-level template code
  File "/usr/local/lib/python3.10/site-packages/jinja2/sandbox.py", line 391, in call
    if not __self.is_safe_callable(__obj):
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 2168, in is_safe_callable
    return isinstance(obj, AllStates) or super().is_safe_callable(obj)
  File "/usr/local/lib/python3.10/site-packages/jinja2/sandbox.py", line 275, in is_safe_callable
    getattr(obj, "unsafe_callable", False) or getattr(obj, "alters_data", False)
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 1959, in _fail_with_undefined_error
    raise ex
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 1950, in _fail_with_undefined_error
    return super()._fail_with_undefined_error(*args, **kwargs)
jinja2.exceptions.UndefinedError: 'None' has no attribute 'weekday'

**Comment by Hellowlol** [comment]
Does the sensor work? The only place your should get this error is in the validation of the template in the gui

**Comment by MS-Turbo** [comment]
Same here. I have two sensors. The sensor written i yaml with additional cost stopped working, but the sensor without cost in integrations is working

**Comment by SVH-Powel** [comment]
Same here:

2022-11-11 06:49:19.678 ERROR (MainThread) [homeassistant.helpers.template] Template variable error: 'None' has no attribute 'isoweekday' when rendering '{% set s = {
"hourly_fixed_cost": 0.0,
"night": 0.2926,
"day": 0.4176,
"cert": 0.01
}
%}
{% if now().isoweekday() >= 1 and now().isoweekday() <= 5 %}
{% if now().hour >=6 and now().hour <22 %}
{{(s.day+s.hourly_fixed_cost+s.cert)|float(0)}}
{% else %}
{{(s.night+s.hourly_fixed_cost+s.cert)|float(0)}}
{% endif %}
{% else %}
{{(s.night+s.hourly_fixed_cost+s.cert)|float(0)}}
{% endif %}'

**Comment by BigFoot2020** [comment]
Had same issue, so I moved time based condition that used now().hour to a separate binary sensor and when using that in additional_cost template, sensor works. 

Likely related to this update: https://github.com/custom-components/nordpool#addition-costs 
now()- this always refer to the current hour of the price

**Comment by toredash** [comment]
> Does the sensor work? The only place your should get this error is in the validation of the template in the gui

Does this mean that the additional_cost template can only be configured via GUI now ?

> Same here. I have two sensors. The sensor written i yaml with additional cost stopped working, but the sensor without cost in integrations is working

I had the same issue when running master-branch 2-3 weeks ago, before I reverted to 0.0.7 release.

I think this is related to this commit: https://github.com/custom-components/nordpool/commit/ee83184a3f47850595f47281157bd51962281e34

Basing it on this comment that was removed as part of the commit
```
# Used to inject the current hour.
# so template can be simplified using now
```

**Comment by Hellowlol** [comment]
Hi!

I know where the bug is, this was a simple oversight on my end, I forgot to account for prices that don't have an hour. Like min, max, average etc.

**Comment by Hellowlol** [comment]
Should be fixed in 0.0.9
